src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardElement.querySelector(".card__title").textContent = data.name;
+
+  setUpImagePreview(cardImageEl, data);
+  setUpLikeButton(cardElement.querySelector(".card__like-btn"), data);
+  cardElement
+    .querySelector(".card__delete-btn")
+    .addEventListener("click", () => openDeleteModal(cardElement, data._id));
+
+  return cardElement;
+}
+
+// Helper function to set up image preview
+function setUpImagePreview(imageElement, data) {
+  imageElement.addEventListener("click", () => {
+    const modalImage = previewModal.querySelector(".modal__image");
+    const modalCaption = previewModal.querySelector(".modal__caption");
+
+    modalImage.src = data.link;
+    modalImage.alt = data.name;
+    modalCaption.textContent = data.name;
+
+    openModal(previewModal);
+  });
+}
+
+// Function for Like button handler
+function setUpLikeButton(likeButton, data) {
+  if (data.isLiked) likeButton.classList.add("card__like-btn_liked");
+
+  likeButton.addEventListener("click", () => {
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+    api
+      .handleLikeStatus(data._id, isLiked)
+      .then((updatedData) =>
+        likeButton.classList.toggle("card__like-btn_liked", updatedData.isLiked)
+      )
+      .catch(console.error);
+  });
+}
+
+// Function for Delete modal logic
+function openDeleteModal(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  const deleteButton = deleteForm.querySelector(".modal__submit-btn");
+  showLoading(deleteButton, false, "Deleting..", "Delete"); // Reset button text to "Delete"
+
+  // Define a new delete handler for the delete action
+  const newDeleteHandler = (evt) => {
+    evt.preventDefault();
+    showLoading(deleteButton, true, "Deleting...", "Delete"); // Show "Deleting..." while processing
+
+    api
+      .deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove(); // Remove the selected card from DOM upon successful deletion
+        closeModal(deleteModal);
+      })
+      .catch((error) => console.error("Failed to delete card:", error))
+      .finally(() => showLoading(deleteButton, false, "Deleting", "Delete")); // Reset button text to "Delete"
+  };
+
+  // Remove any previously attached handler from deleteForm if it exists
+  deleteForm.removeEventListener("submit", deleteForm._deleteHandler);
+
+  // Store the new handler in a property and attach it to the delete form
+  deleteForm._deleteHandler = newDeleteHandler;
+  deleteForm.addEventListener("submit", newDeleteHandler, { once: true });
+
+  // Open the delete modal
+  openModal(deleteModal);
+}
+
+// Profile edit form submission
+editProfileForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = editProfileForm.querySelector(".modal__submit-btn");
```

Comment:
In your theory and videos you have ` const saveButton = evt.submitter;` as the save button. It's very easy to get from `evt`. You don't need to search them every time you submit a form


**Please, correct it everywhere in the code**